internal/pipeline/pipeline.go

```diff
@@ -79,14 +82,48 @@ func (p *Pipeline) String() string {
 	return fmt.Sprintf("%+v", *p)
 }
 
+// Validate will validate all fields in the Pipeline `p`
```

Comment:
nit: You don't need to refer to the parameter `p`. This is a receiver method, so it's obvious it acts on Package. Same for other method receivers.